clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1776,27 +1783,29 @@ void handleFailure(Throwable throwable) {
     public DeleteAclsResult deleteAcls(Collection<AclBindingFilter> filters, DeleteAclsOptions options) {
         final long now = time.milliseconds();
         final Map<AclBindingFilter, KafkaFutureImpl<FilterResults>> futures = new HashMap<>();
-        final List<AclBindingFilter> filterList = new ArrayList<>();
+        final List<DeleteAclsRequestData.DeleteAclsFilter> filterList = new ArrayList<>();
         for (AclBindingFilter filter : filters) {
             if (futures.get(filter) == null) {
-                filterList.add(filter);
+                filterList.add(DeleteAclsRequest.deleteAclsFilter(filter));
                 futures.put(filter, new KafkaFutureImpl<>());
             }
         }
+        final DeleteAclsRequestData data = new DeleteAclsRequestData().setFilters(filterList);
         runnable.call(new Call("deleteAcls", calcDeadlineMs(now, options.timeoutMs()),
             new LeastLoadedNodeProvider()) {
 
             @Override
             DeleteAclsRequest.Builder createRequest(int timeoutMs) {
-                return new DeleteAclsRequest.Builder(filterList);
+                return new DeleteAclsRequest.Builder(data);
             }
 
             @Override
             void handleResponse(AbstractResponse abstractResponse) {
                 DeleteAclsResponse response = (DeleteAclsResponse) abstractResponse;
                 List<AclFilterResponse> responses = response.responses();
                 Iterator<AclFilterResponse> iter = responses.iterator();
-                for (AclBindingFilter filter : filterList) {
+                //FIXME What about duplicates?
```

Comment:
This is another case that needs to be fixed.  The simplest way would be just creating a list which contains all the AclBindingFilter objects that were sent on the wire (it seems that the incorrect decision to use `filters` in this PR was caused by the change in type in `filterList` from `AclBindingFilter` to `DeleteAclsRequestData.DeleteAclsFilter`)

Reply:
Again, this was not complete and hence why there was a FIXME in it. ;)